packages/analytics/amplify_analytics_pinpoint_dart/test/queued_item_store_test.dart

```diff
@@ -0,0 +1,299 @@
+      expect(readValues, values.sublist(0, 7));
+    });
+
+    test('deletes first, middle, and last subsets of stored items', () async {
+      const values = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+      for (final value in values) {
+        await db.add(value);
+      }
+
+      const valuesToDelete = ['0', '1', '4', '5', '8', '9'];
+      final itemsToDelete = (await db.get(100)).where((item) {
+        return valuesToDelete.contains(item.value);
+      });
+      await db.deleteItems(itemsToDelete);
+
+      final readItems = await db.get(100);
+      final readValues = readItems.map((e) => e.value);
+      expect(readValues, const ['2', '3', '6', '7']);
+    });
+
+    test('deletes inner left and right subsets of stored items', () async {
+      const values = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+      for (final value in values) {
+        await db.add(value);
+      }
+
+      const valuesToDelete = ['1', '2', '4', '5', '7', '8'];
+      final itemsToDelete = (await db.get(100)).where((item) {
+        return valuesToDelete.contains(item.value);
+      });
+      await db.deleteItems(itemsToDelete);
+
+      final readItems = await db.get(100);
+      final readValues = readItems.map((e) => e.value);
+      expect(readValues, const ['0', '3', '6', '9']);
+    });
+
+    test('deletes the first stored item', () async {
+      const values = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+      for (final value in values) {
+        await db.add(value);
+      }
+
+      final itemsToDelete = (await db.get(100)).toList().sublist(0, 1);
+      await db.deleteItems(itemsToDelete);
+
+      final readItems = await db.get(100);
+      final readValues = readItems.map((e) => e.value);
+      expect(readValues, values.sublist(1));
+    });
+
+    test('deletes the last stored item', () async {
+      const values = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+      for (final value in values) {
+        await db.add(value);
+      }
+
+      final itemsToDelete = (await db.get(100)).toList().sublist(9);
+      await db.deleteItems(itemsToDelete);
+
+      final readItems = await db.get(100);
+      final readValues = readItems.map((e) => e.value);
+      expect(readValues, values.sublist(0, 9));
+    });
+
+    test('deletes every item with an odd index', () async {
+      const values = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];
+      for (final value in values) {
+        await db.add(value);
+      }
+
+      const valuesToDelete = ['1', '3', '5', '7', '9'];
+      final itemsToDelete = (await db.get(100)).where((item) {
+        return valuesToDelete.contains(item.value);
+      });
+      await db.deleteItems(itemsToDelete);
+
+      final readItems = await db.get(100);
+      final readValues = readItems.map((e) => e.value);
+      expect(readValues, const ['0', '2', '4', '6', '8']);
+    });
+
+    test('deletes every item with an even index', () async {
```

Comment:
*nit:* Some of the tests for delete seem redundant. I appreciate the desire to be thorough and capture all edge cases. However, I am not sure each of these is testing something unique.

These seem like the core test cases.
- Can delete a single item (first, last, middle)
- Can delete all items
- Can delete a range (first, last, middle)
- Can delete multiple ranges
- Does nothing if items were previously deleted - could test this in various ways (ranges, single deletes) but not sure that is really needed.

I think you covered those, but not sure that some of the tests are really different than others.

Reply:
Okay thanks - I removed 4 tests:
even, odd, delete & save x2 tests